### ISSUE TITLE
Serialization to XML?

### ISSUE THREAD

**Issue by bplett** [angularsen/UnitsNet#270]
This is a question, not an issue, but I don't know a better way to ask.  Is there no default serialization of units?  For example, I have a property in my class defined as `public Length PipeDiameter { get; set; }`.  When I serialize to XML, I get `<PipeDiameter />`.  How do I serialize this properly?  I really don't want to write my own conversion property if I can avoid it.

I've noticed there appears to be JSON serialization support.  Is there no support for XML serialization?

Thanks!
Brad.

**Comment by anjdreas** [comment]
Hi, you are right there is no support for default serialization. This is partly due to the immutable design of the types, but we also decided to not support default serialization as we [can't guarantee backwards compatibility over time](https://github.com/anjdreas/UnitsNet/#serialization). This means that what you serialize today may not deserialize in the future or on another system running a different version of Units.NET. There are many examples of breaking changes in the history already if we had supported serialization, due to changing the base units of quantities. We specifically wrote a JSON serializer to be more robust by serializing both the base unit name and the value of the base unit representation. This way, unless that unit disappears later, it will be able to deserialize.

I suppose the solution here is to write a similar kind of serializer to support XML and binary serialization out of the box, but no plans for that as of now. If you want to take a stab at it, you are most welcome to start a PR and we can help out with our experience doing JSON.

https://github.com/anjdreas/UnitsNet/tree/master/UnitsNet.Serialization.JsonNet

cc @bretthysuik @neutmute @eriove
